### ISSUE TITLE
How http sender's failures are handled?

### ISSUE THREAD

**Issue by DrakkLord** [ACRA/acra#562]
As far as I know the usual flow of ACRA is : crash detected ->collect crash data -> spawn another process -> new process sends report.
I'm aware that reports are kept for later retries, so I assume here if a send fails then ACRA will retry some later time I guess on next startup.
I'm using ACRA in an Android Wear watch face, the flow of ACRA I think assumes that once in a while the app is restarted, however a watchface can be a very long running application which may not be restarted for days. If I assume that the report sender tries X times on the first try then tries again on the next application start then likely that I may not get the report for days.

So to sum it up:
- how many times does ACRA tries to send a report if it fails to send it in the first try?
- what are the triggers for re-sending reports? Application startup == ACRA initialization?
- is there function that would allow trigger sending unsent reports when the client application sees fit?

**Comment by F43nd1r** [comment]
> how many times does ACRA tries to send a report if it fails to send it in the first try?

ACRA does not try to send reports multiple times in one run.

> what are the triggers for re-sending reports? Application startup == ACRA initialization?

By default, yes.

> is there function that would allow trigger sending unsent reports when the client application sees fit?

`ErrorReporter.checkReportsOnApplicationStart()` named this way for legacy reasons.

**Comment by DrakkLord** [comment]
Okay so if understand correctly, ACRA will try to send the report once, then retry on each application startup + I can call ErrorReporter.checkReportsOnApplicationStart() to trigger manually?
Can I call that function rapidly or say randomly without causing any errors? Say I call it once but then call it again while a report is currently being sent.

**Comment by F43nd1r** [comment]
> Okay so if understand correctly, ACRA will try to send the report once, then retry on each application startup + I can call ErrorReporter.checkReportsOnApplicationStart() to trigger manually?

Yes

> Can I call that function rapidly or say randomly without causing any errors? Say I call it once but then call it again while a report is currently being sent.

Multiple parallel calls should be no problem for the service, but I have never tested it, so no guarantees.
